src/Shared/Tests/CompilerAssert.cs

```diff
@@ -4,11 +4,36 @@
 using Microsoft.CodeAnalysis.Testing;
 using Microsoft.CodeAnalysis.Testing.Verifiers;
 using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Shared.Tests;
 
 public static class CompilerAssert
 {
+    /// <summary>
+    /// Compile the file associated with he caller's file name.
+    /// </summary>
+    /// <param name="expectedErrorIds"></param>
+    /// <returns></returns>
+    public static async Task CompileTestTargetFileAsync(
+        string[] expectedErrorIds, [CallerFilePath] string testFileName = null!)
+    {
+        await CompileAsync(new string[] {
+            GetTargetFileNameToCompileFromTestFileName(testFileName) }, 
+            expectedErrorIds);
+    }
+    
+    public static string GetTargetFileNameToCompileFromTestFileName(
+        [CallerFilePath] string testFileName = null!)
+    {
+#if NET7_0_OR_GREATER
+        ArgumentException.ThrowIfNullOrEmpty(testFileName);
+#endif
```

Comment:
Should there be an #else that also does the null check?

Reply:
I decided not to do this because we would just have to remove when we drop .NET 6 , it and it wasn't germane anyway.